packages/big-design/src/components/AccordionGroup/spec.tsx

```diff
@@ -0,0 +1,113 @@
+import { ErrorIcon } from '@bigcommerce/big-design-icons';
+import userEvent from '@testing-library/user-event';
+import React from 'react';
+
+import { render, screen, waitFor } from '@test/utils';
+
+import { Text } from '../Typography';
+
+import { AccordionGroup } from './AccordionGroup';
+import { AccordionPanel } from './AccordionPanel';
+
+const handleClick = jest.fn();
+
+const mockAccordion = (
+  <AccordionPanel header="Panel Header" isExpanded onClick={handleClick}>
+    <Text>This is a child component</Text>
+  </AccordionPanel>
+);
+
+const mockAccordionGroup = (
+  <AccordionGroup header="Test Accordion">
+    <AccordionPanel header="Panel Header" isExpanded onClick={handleClick}>
+      <Text>This is a child component</Text>
+    </AccordionPanel>
+  </AccordionGroup>
+);
+
+test('it renders accordion group header', async () => {
+  render(mockAccordionGroup);
+
+  const header = await screen.findByRole('heading', { name: /test accordion/i });
+
+  expect(header).toMatchSnapshot();
+  expect(header).toBeVisible();
+});
+
+test('it renders accordion children', async () => {
+  render(mockAccordionGroup);
+
+  const child = await screen.findByText('Panel Header');
+
+  expect(child).toBeVisible();
+});
+
+test('accordion item renders header', async () => {
+  render(mockAccordion);
+
+  const header = await screen.findByText('Panel Header');
+
+  expect(header).toBeVisible();
+});
+
+test('accordion row collapses and expands on click', async () => {
+  render(mockAccordion);
+
+  const panel = await screen.findByRole('button');
+
+  userEvent.click(panel);
+
+  await waitFor(() => {
+    expect(panel.getAttribute('aria-expanded')).toBe('true');
+  });
+
+  userEvent.click(panel);
+
+  await waitFor(() => {
+    expect(panel.getAttribute('aria-expanded')).toBe('false');
+  });
```

Comment:
`panel` seem to refer to the button 🤔 maybe rename that?

Reply:
Also, to fix the issues you were seeing in CI, we should rewrite these tests to be a bit more semantic instead of using `waitFor`...
```suggestion
  const button = await screen.findByRole('button');

  userEvent.click(button);
  
  const panel = await screen.findByRole('region');
  
  expect(panel).toBeVisible();

  userEvent.click(button);

  expect(panel).not.toBeVisible();
```

Might have to tinker with some `findByRole` options, but I believe that should be a bit more meaningful. You should probably update the other tests as well.

Reply:
Updated the tests but still seeing a failure for the second assertion - https://github.com/bigcommerce/big-design/pull/933#issuecomment-1211032183

Reply:
Added @MariaJose's suggestion (https://github.com/bigcommerce/big-design/pull/933#issuecomment-1203259014) to fix it